lib/std/os.zig

```diff
@@ -455,7 +455,44 @@ pub fn abort() noreturn {
     if (!builtin.link_libc and builtin.os.tag == .linux) {
         raise(SIG.ABRT) catch {};
 
-        // TODO the rest of the implementation of abort() from musl libc here
+        // If there was a SIGABRT handler installed and it returned, or if
+        // SIGABRT was blocked or ignored, take an Async-Signal-safe lock to
+        // prevent sigaction from installing a new SIGABRT handler, uninstall
+        // any handler that may be present, and re-raise the signal to generate
+        // the default action of abnormal termination.
+        // TODO handle failures on setting sigprocmask
+        _ = linux.sigprocmask(SIG.BLOCK, &linux.all_mask, null);
+
+        // TODO decide where (conditionally) abort_lock mutex should go
```

Comment:
Not sure, where global variables/state of zig binaries should go.
Are there meta functions to access them directly or via offset?

Reply:
added TODO for the atomic to use.